lib/logger/lib/logger/legacy_handler.ex

```diff
@@ -0,0 +1,180 @@
+defmodule Logger.LegacyHandler do
+  @moduledoc false
+
+  @doc false
+  def child_spec(_) do
+    %{
+      id: :gen_event,
+      start: {:gen_event, :start_link, [{:local, Logger}]},
+      modules: :dynamic
+    }
+  end
+
+  @doc """
+  Hook required by `:logger`.
+  """
+  def log(%{level: erl_level, msg: msg, meta: erl_meta}, config) do
+    level = erlang_level_to_elixir_level(erl_level)
+
+    case Logger.Config.log_data(level) do
```

Comment:
So it seems we are checking this twice now: once when we call `Logger.log` and again here. We should probably have a metadata that checks if it came from `Logger` and avoid doing those checks again (if it came from Logger).

Reply:
Yes, I have fixed that locally. I need to split it into logical parts and I will commit in a second.